### PR TITLE
fix(webjob): test coverage for _executionTasks=true requests without file upload

### PR DESCRIPTION
**Pull request by mulahasanovic**
## Issue #882 

Viya rejects requests posted to _executionTasks=true jobs with 502 or 403 when the adapter sends `Content-Type: multipart/form-data` and config set to `useComputeApi=null` (or `undefined`).

## Intent

Resolve empty form data payloads.

Update: An issue was confirmed on all multipart form data uploads that do not include an actual file upload with executionTasks flag set to true.

## Implementation

- ~override added when empty form-data is sent, set `text/plain` content type~
- ~use url encoded payload when executionTasks flag is set, without a file~
- add test coverage
- add regression test, and failing test for non-empty form-data payload

## Checks

No PR (that involves a non-trivial code change) should be merged, unless all items below are confirmed!  If an urgent fix is needed - use a tar file.

- [x] Unit tests coverage has been increased and a new threshold is set.
- [x] All `sasjs-cli` unit tests are passing (`npm test`).
- (CI Runs this) All `sasjs-tests` are passing. If you want to run it manually (instructions available [here](https://github.com/sasjs/adapter/blob/master/sasjs-tests/README.md)).
- [ ] [Data Controller](https://datacontroller.io) builds and is functional on both SAS 9 and Viya

### CODE REVIEW

File: src/job-execution/spec/WebJobExecutor.spec.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+import { ServerType } from '@sasjs/utils/types'
+import { WebJobExecutor } from '../WebJobExecutor'
+import { RequestClient } from '../../request/RequestClient'
+import { SASViyaApiClient } from '../../SASViyaApiClient'
+
+describe('WebJobExecutor.execute() Content-Type selection', () => {
+  const serverUrl = 'https://sample.server.com'
+  const jobsPath = '/SASJobExecution'
+
+  const makeExecutor = (serverType: ServerType = ServerType.Sas9) => {
```
Comment (allanbowe) [inline src/job-execution/spec/WebJobExecutor.spec.ts]: should this be set to Viya, rather than Sas9?

File: src/job-execution/spec/WebJobExecutor.spec.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
+import NodeFormData from 'form-data'
+import { ServerType } from '@sasjs/utils/types'
+import { WebJobExecutor } from '../WebJobExecutor'
+import { RequestClient } from '../../request/RequestClient'
+import { SASViyaApiClient } from '../../SASViyaApiClient'
+
+describe('WebJobExecutor.execute() Content-Type selection', () => {
+  const serverUrl = 'https://sample.server.com'
```
Comment (allanbowe) [inline src/job-execution/spec/WebJobExecutor.spec.ts]: should this be left empty?

File: src/job-execution/WebJobExecutor.ts [hunk 3]
```diff
@@ -107,22 +107,38 @@ export class WebJobExecutor extends BaseJobExecutor {
...
+      (config.serverType === ServerType.Sas9 ||
+        stringifiedData.length > 500000 ||
+        stringifiedData.includes(';'))
+
+    // Viya rejects empty multipart on _executionTasks=true when no file is
+    // uploaded. Use x-www-form-urlencoded in that case; URL params stay.
+    const parsedUrl = new URL(apiUrl)
+    const isExecutionTasksEndpoint =
+      (parsedUrl.searchParams.has('_program') ||
+        parsedUrl.searchParams.has('__program')) &&
```
Comment (allanbowe) [inline src/job-execution/WebJobExecutor.ts]: __program should only happen when submitting JES API approach, not web

File: src/job-execution/spec/WebJobExecutor.spec.ts (hunk 1 above)
Comment (mulahasanovic) [reply src/job-execution/spec/WebJobExecutor.spec.ts]: Sas9 was a cheap way to force the file upload branch without needing `;` or large payload, and a quick way to increase coverage and hit more code branches.

We can drop Sas9 to keep PR scope cleaner.

File: src/job-execution/spec/WebJobExecutor.spec.ts (hunk 2 above)
Comment (mulahasanovic) [reply src/job-execution/spec/WebJobExecutor.spec.ts]: We needed some value for `serverUrl` to mock requests, we used the same "sample server" in `FileUploader` spec tests.

File: src/job-execution/spec/WebJobExecutor.spec.ts (hunk 1 above)
Comment (allanbowe) [reply src/job-execution/spec/WebJobExecutor.spec.ts]: no need, more coverage is great, I missed that it was a spec file
